packages/perseus/src/util/widget-enum-utils.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Some helpers for CEDAR logging. CEDAR convention dictates
+ * that enum values should be upper case. Ex: for the
+ * PerseusWidgetUsedInProblemAttempt CEDAR event, we needed
+ * to: get the type and convert that to an enum value.
+ */
+
+type WidgetName =
+    | "categorizer"
+    | "cs-program"
+    | "definition"
+    | "deprecated-standin"
+    | "dropdown"
+    | "explanation"
+    | "expression"
+    | "graded-group-set"
+    | "graded-group"
+    | "grapher"
+    | "group"
+    | "iframe"
+    | "image"
+    | "input-number"
+    | "interaction"
+    | "interactive-graph"
+    | "label-image"
+    | "lights-puzzle"
+    | "matcher"
+    | "matrix"
+    | "measurer"
+    | "molecule-renderer"
+    | "number-line"
+    | "numeric-input"
+    | "orderer"
+    | "passage"
+    | "passage-ref"
+    | "passage-ref-target"
+    | "plotter"
+    | "python-program"
+    | "reaction-diagram"
+    | "radio"
+    | "sequence"
+    | "simulator"
+    | "sorter"
+    | "table"
+    | "unit-input"
+    | "video";
+
+type WidgetEnum =
+    | "CATEGORIZER"
+    | "CS_PROGRAM"
+    | "DEFINITION"
+    | "DEPRECATED_STANDIN"
+    | "DROPDOWN"
+    | "EXPLANATION"
+    | "EXPRESSION"
+    | "GRADED_GROUP"
+    | "GRADED_GROUP_SET"
+    | "GRAPHER"
+    | "GROUP"
+    | "IFRAME"
+    | "IMAGE"
+    | "INPUT_NUMBER"
+    | "INTERACTIVE"
+    | "INTERACTIVE_GRAPH"
+    | "LABEL_IMAGE"
+    | "LIGHTS_PUZZLE"
+    | "MATCHER"
+    | "MATRIX"
+    | "MEASURER"
+    | "MOLECULE"
+    | "NUMBER_LINE"
+    | "NUMERIC_INPUT"
+    | "ORDERER"
+    | "PASSAGE"
+    | "PASSAGE_REF"
+    | "PASSAGE_REF_TARGET"
+    | "PLOTTER"
+    | "PYTHON_PROGRAM"
+    | "REACTION_DIAGRAM"
+    | "RADIO"
+    | "SEQUENCE"
+    | "SIMULATOR"
+    | "SORTER"
+    | "TABLE"
+    | "UNIT"
+    | "VIDEO";
+
+const widgetNameToEnum: Record<WidgetName, WidgetEnum> = {
+    categorizer: "CATEGORIZER",
+    "cs-program": "CS_PROGRAM",
+    definition: "DEFINITION",
+    "deprecated-standin": "DEPRECATED_STANDIN",
+    dropdown: "DROPDOWN",
+    explanation: "EXPLANATION",
+    expression: "EXPRESSION",
+    "graded-group-set": "GRADED_GROUP",
+    "graded-group": "GRADED_GROUP_SET",
```

Comment:
Are these reversed?

Reply:
Oh that's weird. That's copy/pasted from prod webapp which means it's live on the site. 🤔 Not that I think we care all that much about graded-group statistics (compared to more interactive widgets).